src/document.rs

```diff
@@ -1,27 +1,204 @@
 use serde::{Deserialize, Serialize};
+use std::fmt::Debug;
 
+/// Options for document insertion.
 #[derive(Serialize, Deserialize, Debug)]
-pub struct Document<T> {
+#[serde(rename_all = "camelCase")]
+pub struct DocumentInsertOptions {
+    /// Wait until document has been synced to disk.
+    pub wait_for_sync: Option<bool>,
```

Comment:
What's the point to wrap all field into `Option<Type>`? If I am correct, that's becuase all fields are optonal. We can even go with every field set to `None`.

If this is the case, I suggest use a builder pattern and mark every field as private to avoid access. Use [`typed_builder`](https://docs.rs/typed-builder/0.6.0/typed_builder/index.html) to help construct a builder type for this struct.

```rust
#[derive(TypedBuilder)]
struct DocumentInsertOptions{
    // #[default] without parameter - use the type's default
    // #[builder(setter(strip_option))] - wrap the setter argument with `Some(...)`
    #[builder(default, setter(strip_option))]
    wait_for_sync: Option<bool>,
     // .....
}
```

With the above type, we can set what we need only, like `DocumentInsertOptions::builder().silent(true).wait_for_async(true).build()`.